src/v/cluster/cluster_utils.h

```diff
@@ -385,4 +388,17 @@ get_allocation_domain(const model::ntp& ntp) {
 partition_state get_partition_state(ss::lw_shared_ptr<cluster::partition>);
 partition_raft_state get_partition_raft_state(consensus_ptr);
 
+/**
+ * Check that the configuration is valid, if not return a string with the
+ * error cause.
+ *
```

Comment:
nit: may be good to add a line for context of why we only compare the changed parts of the configuration.

Reply:
done